### PR TITLE
Fix regexes for Ruby

### PR DESCRIPTION
**Pull request by sanchezzzhak**
#6991

### CODE REVIEW

File: regexes/client/browsers.yml [hunk 1]
```diff
@@ -2420,7 +2420,7 @@
   version: '$1'
 
 # MAUI Wap Browser
-- regex: 'WAP Browser/MAUI|(?:\w+)?Maui Wap Browser|MAUI[- ]Browser'
+- regex: 'WAP Browser/MAUI|(?:.*_)?Maui Wap Browser|MAUI[- ]Browser'
```
Comment (sgiehl) [inline regexes/client/browsers.yml:2423]: Doesn't that actually change the regex? Wouldn't that change be needed to keep that same behaviour?
```suggestion
- regex: 'WAP Browser/MAUI|(?:\w*)Maui Wap Browser|MAUI[- ]Browser'
```

File: regexes/device/mobiles.yml [hunk 2]
```diff
@@ -24160,7 +24160,7 @@ Beelink:
 
 # MAXVI
 MAXVI:
-  regex: 'MAXVI[ _](?:MS401)(?:[);/ ]|$)|(?:\w+)?_MAXVI_\w+|MS502[_ ]Orion|MS531'
+  regex: 'MAXVI[ _](?:MS401)(?:[);/ ]|$)|.*_?MAXVI?_\w|MS502[_ ]Orion|MS531'
```
Comment (sgiehl) [inline regexes/device/mobiles.yml:24163]: That doesn't seem to be the same as before either:
```suggestion
  regex: 'MAXVI[ _](?:MS401)(?:[);/ ]|$)|\w*_?MAXVI?_\w+|MS502[_ ]Orion|MS531'
```

File: regexes/device/mobiles.yml [hunk 3]
```diff
@@ -24169,7 +24169,7 @@ MAXVI:
       model: 'MS531 Vega'
     - regex: 'MS502[_ ]Orion'
       model: 'MS502 Orion'
-    - regex: '(?:\w+)?MAXVI_(P11|K10|X[358]00)_\w+'
+    - regex: '.*_?MAXVI_(P11|K10|X[358]00)_'
```
Comment (sgiehl) [inline regexes/device/mobiles.yml:24172]:
```suggestion
    - regex: '\w*MAXVI_(P11|K10|X[358]00)_'
```

File: regexes/oss.yml [hunk 4]
```diff
@@ -139,23 +139,23 @@
...
+- regex: 'Pinterest for Android/.*; (\d(?:[\d\.]*))[)]$'
   name: 'Android'
   version: '$1'
 
 - regex: 'Android; (\d+[\.\d]*); Mobile;'
   name: 'Android'
   version: '$1'
 
-- regex: '[ ]([\d\.]+)\) AppleWebKit.*ROBLOX Android App'
+- regex: '[ ]([\d\.]+)[)] AppleWebKit.*ROBLOX Android App'
```
Comment (sgiehl) [inline regexes/oss.yml:158]: Didn't the escaping work for ruby?

**Review by sgiehl — CHANGES_REQUESTED** [review]
Left some comments, where the changes actually were not what I would have expected.

**Review by sgiehl — APPROVED** [review]

**Comment by jrochkind** [comment]
thank you! --a rubyist

**Pull request merged**